core/sessions/orm.go

```diff
@@ -0,0 +1,192 @@
+package sessions
+
+import (
+	"crypto/subtle"
+	"database/sql"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/auth"
+	"github.com/smartcontractkit/chainlink/core/bridges"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
+)
+
+type ORM interface {
+	FindUser() (User, error)
+	AuthorizedUserWithSession(sessionID string) (User, error)
+	DeleteUser() error
+	DeleteUserSession(sessionID string) error
+	CreateSession(sr SessionRequest) (string, error)
+	ClearNonCurrentSessions(sessionID string) error
+	CreateUser(user *User) error
+	SetAuthToken(user *User, token *auth.Token) error
+	DeleteAuthToken(user *User) error
+	SetPassword(user *User, newPassword string) error
+	Sessions(offset, limit int) ([]Session, error)
+
+	FindExternalInitiator(eia *auth.Token) (initiator *bridges.ExternalInitiator, err error)
+}
+
+type orm struct {
+	db              *sqlx.DB
+	sessionDuration time.Duration
+}
+
+var _ ORM = (*orm)(nil)
+
+func NewORM(db *sqlx.DB, sessionDuration time.Duration) ORM {
+	return &orm{db, sessionDuration}
+}
+
+// FindUser will return the one API user, or an error.
+func (o *orm) FindUser() (User, error) {
+	return o.findUser()
+}
+
+func (o *orm) findUser() (user User, err error) {
+	sql := "SELECT * FROM users ORDER BY created_at desc LIMIT 1"
+	err = o.db.Get(&user, sql)
+	return
+}
+
+// AuthorizedUserWithSession will return the one API user if the Session ID exists
+// and hasn't expired, and update session's LastUsed field.
+func (o *orm) AuthorizedUserWithSession(sessionID string) (User, error) {
+	if len(sessionID) == 0 {
+		return User{}, errors.New("Session ID cannot be empty")
+	}
+
+	result, err := o.db.Exec("UPDATE sessions SET last_used = now() WHERE id = $1 AND last_used + $2 >= now()", sessionID, o.sessionDuration)
+	if err != nil {
+		return User{}, err
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return User{}, err
+	}
+	if rowsAffected == 0 {
+		return User{}, sql.ErrNoRows
+	}
+	return o.FindUser()
+}
+
+// DeleteUser will delete the API User in the db.
+func (o *orm) DeleteUser() error {
+	ctx, cancel := postgres.DefaultQueryCtx()
+	defer cancel()
+	return postgres.SqlxTransaction(ctx, o.db, func(tx *sqlx.Tx) error {
+		// sql := "DELETE FROM users ORDER BY created_at DESC LIMIT 1"
```

Comment:
?

Reply:
`DELETE FROM .. ORDER BY .. LIMIT` doesn't work on postgres, but the code seems to always expect a single DB user (that's why FindUser and DeleteUser take no arguments) and clears out all existing sessions afterwards. I just wipe all the users with `DELETE FROM` now.

Reply:
You can use a nested select to do it